radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/api/actors/ApiTxnSubmitter.java

```diff
@@ -0,0 +1,254 @@
+		if (tokenTypes.isEmpty()) {
+			return null;
+		} else {
+			var nextIndex = random.nextInt(tokenTypes.size());
+			return tokenTypes.get(nextIndex);
+		}
+	}
+
+	private MintTokens mintTokens(NodeApiClient nodeClient, EntityIdentifier to, Random random) throws Exception {
+		var tokenResourceIdentifier = findTokenClientOwns(nodeClient, random);
+		if (tokenResourceIdentifier == null) {
+			return null;
+		}
+
+		return new MintTokens(nextAmount(random), tokenResourceIdentifier, to);
+	}
+
+	private BurnTokens burnTokens(NodeApiClient nodeClient, Random random) throws Exception {
+		var tokenResourceIdentifier = findTokenClientOwns(nodeClient, random);
+		if (tokenResourceIdentifier == null) {
+			return null;
+		}
+		var publicKey = nodeClient.getPublicKey();
+		var accountIdentifier = nodeClient.deriveAccount(publicKey);
+		return new BurnTokens(nextAmount(random), tokenResourceIdentifier, accountIdentifier);
+	}
+
+	private CreateTokenDefinition createTokenDefinition(NodeApiClient nodeClient, EntityIdentifier to, Random random) throws Exception {
+		var publicKey = nodeClient.getPublicKey();
+		var owner = nodeClient.deriveAccount(publicKey);
+		var uint256Bytes = new byte[UInt256.BYTES];
+		random.nextBytes(uint256Bytes);
+		var amount = UInt256.from(uint256Bytes);
+		if (random.nextBoolean()) {
+			return CreateTokenDefinition.mutableTokenSupply("test" + tokenId++, amount, owner, to);
+		} else {
+			return CreateTokenDefinition.fixedTokenSupply("test" + tokenId++, amount, to);
+		}
+	}
+
+	@Override
+	public String execute(MultiNodeDeterministicRunner runner, Random random) throws Exception {
+		int size = runner.getSize();
+		var nodeIndex = random.nextInt(size);
+		var nodeInjector = runner.getNode(nodeIndex);
+		var nodeClient = nodeInjector.getInstance(NodeApiClient.class);
+		var otherNodeIndex = random.nextInt(size);
+		var otherKey = runner.getNode(otherNodeIndex)
+			.getInstance(Key.get(ECPublicKey.class, Self.class));
+		var next = random.nextInt(11);
+
+		// Don't let the last validator unregister
+		if (next == 4 && nodeIndex <= 0) {
+			return "Skipped";
+		}
+
+		NodeTransactionAction action = switch (next) {
+			case 0 -> transferTokens(nodeClient, nodeClient.deriveAccount(otherKey), random);
+			case 1 -> new StakeTokens(nextAmount(random), nodeClient.deriveValidator(otherKey));
+			case 2 -> new UnstakeStakeUnits(nextAmount(random), nodeClient.deriveValidator(otherKey).getAddress());
+			case 3 -> new RegisterValidator(true);
+			case 4 -> new RegisterValidator(false);
+			case 5 -> new SetValidatorFee(random.nextInt(ValidatorUpdateRakeConstraintScrypt.RAKE_MAX + 1));
+			case 6 -> new SetValidatorOwner(nodeClient.deriveAccount(otherKey));
+			case 7 -> new SetAllowDelegationFlag(random.nextBoolean());
+			case 8 -> createTokenDefinition(nodeClient, nodeClient.deriveAccount(otherKey), random);
+			case 9 -> mintTokens(nodeClient, nodeClient.deriveAccount(otherKey), random);
+			case 10 -> burnTokens(nodeClient, random);
```

Comment:
Could we make those numbers more descriptive?

Reply:
Could turn these into enums but imo that's more levels of indirection than necessary